output mean+quantiles table + transmission spectrum

cpiaulet opened the pull request:
- results table for each parameter: mean, 16th, 50th, 84th percentiles, +1 and -1sigma (calculated traditionally as 84th-50th and 16th-50th)
- transmission spectrum table
- tested for emcee, dynesty, lsq, for white and wavelength-dependent light curves + shared fits

taylorbell57 reviewed (changes requested):
Much of this functionality is already included in Stage 6, so I don't see much benefit of doing so here. I made that Stage 6 code separate from Stage 5 because at the time Kevin had told me to output the transmission spectrum in Stages 6 so that people wouldn't end up biasing their Stage 5 fits as much. Can you explain why you'd rather compute the transmission spectrum here rather than in stage 6 where the information from all wavelengths can be combined into one file? Also, if we're outputting the transmission spectrum in stage 5, then it may be worth copying over chunks of the stage 6 code to output the emission spectrum as well or allow different output units.

cpiaulet commented on eureka_analysis/JWST/run_eureka_demo.py:
```diff
@@ -0,0 +1,18 @@
+import sys
```
I added it to the gitignore, my mistake

taylorbell57 replied:
You will also need to delete the file (using `git rm --cached eureka_analysis/JWST/run_eureka_demo.py` to remove it from GitHub but keep it locally)

cpiaulet commented on eureka/lib/astropytable.py:
```diff
@@ -97,6 +97,7 @@ def savetable_S5(filename, time, wavelength, bin_width, lcdata, lcerr, model, re
     Revisions
     ---------
     """
+
```
fixed + pushed

cpiaulet commented on eureka/S5_lightcurve_fitting/fitters.py:
```diff
@@ -87,7 +93,27 @@ def callback_full(theta, lc, model, prior1, prior2, priortype, freenames):
+    # Save transmission spectrum
+    # indices of fitted rps for all channels
+    ind_spec = np.array([i for i in range(len(freenames)) if 'rp' in freenames[i]])
+    if len(ind_spec):
```
I had implemented it and presented it before for Stage 5 so I thought this was something that people would find useful in this stage as I for instance don't use Stage 6 (for now?) - I started this out because that's an output I needed and didn't have, but it wasn't in S6 at the time most likely. I can see how it's better suited for Stage 6 so I will remove the transmission spectrum code from functions. I'll post here when it's done.

cpiaulet replied:
ok, it's done!

cpiaulet commented on eureka/S5_lightcurve_fitting/fitters.py:
```diff
@@ -928,17 +1007,32 @@ def load_old_fitparams(meta, log, channel, freenames):
-        data = np.append(data, -lower_errs.reshape(1,-1), axis=0)
-        data = np.append(data, upper_errs.reshape(1,-1), axis=0)
-    np.savetxt(meta.outputdir+fname+'.csv', fit_params.reshape(1,-1), header=','.join(freenames), delimiter=',')
+        fname = f'S5_{fitter}_fitparams_ch{lc.channel}'
```
done

cpiaulet commented on eureka/S5_lightcurve_fitting/fitters.py:
```diff
@@ -928,17 +1007,32 @@ def load_old_fitparams(meta, log, channel, freenames):
+        if lc.share:
+            spec_fname = f'S5_{fitter}_trspec_shared'
+        else:
+            spec_fname = f'S5_{fitter}_trspec_ch{lc.channel}'
```
done

taylorbell57 commented on eureka/S5_lightcurve_fitting/fitters.py:
```diff
@@ -87,7 +93,27 @@ def callback_full(theta, lc, model, prior1, prior2, priortype, freenames):
+    t_results = table.Table([freenames, fit_params], 
+                            names=("Parameter", "Mean")  ) 
+
+    # Save transmission spectrum
+    # indices of fitted rps for all channels
+    # ind_spec = np.array([i for i in range(len(freenames)) if 'rp' in freenames[i]])
+    # # get wavelengths at middle of bin
+    # wave_low = meta.wave_low[lc.fitted_channels]
+    # wave_hi = meta.wave_hi[lc.fitted_channels]
+    # wave_mid = (wave_hi+wave_low)/2.
+    # # Save transmission spectrum
+    # # indices of fitted rps for all channels
+    # ind_spec = np.array([i for i in range(len(freenames)) if 'rp' in freenames[i]])
+    # if len(ind_spec):
+    #     t_spec = table.Table([wave_low, wave_mid, wave_hi, fit_params[ind_spec]**2.*1e6, \
+    #                       np.zeros_like(wave_low), np.zeros_like(wave_low)],
+    #                      names=("wave_low_um", "wave_mid_um", "wave_upp_um", "RpRs2_ppm", "err_low_ppm", "err_upp_ppm"))
+    # else:
+        # t_spec = None
```
I would delete all these commented out chunks of code unless @kevin218 disagrees. Having large commented out blocks of code makes things hard to read through in the future. If ever we need the code, it is already saved in the Git history

cpiaulet replied:
done

taylorbell57 commented on eureka/S5_lightcurve_fitting/fitters.py:
```diff
@@ -87,7 +93,27 @@ def callback_full(theta, lc, model, prior1, prior2, priortype, freenames):
+    t_results = table.Table([freenames, fit_params], 
+                            names=("Parameter", "Mean")  ) 
+
+    # Save transmission spectrum
+    # indices of fitted rps for all channels
+    # ind_spec = np.array([i for i in range(len(freenames)) if 'rp' in freenames[i]])
+    # # get wavelengths at middle of bin
+    # wave_low = meta.wave_low[lc.fitted_channels]
+    # wave_hi = meta.wave_hi[lc.fitted_channels]
+    # wave_mid = (wave_hi+wave_low)/2.
+    # # Save transmission spectrum
+    # # indices of fitted rps for all channels
+    # ind_spec = np.array([i for i in range(len(freenames)) if 'rp' in freenames[i]])
+    # if len(ind_spec):
+    #     t_spec = table.Table([wave_low, wave_mid, wave_hi, fit_params[ind_spec]**2.*1e6, \
+    #                       np.zeros_like(wave_low), np.zeros_like(wave_low)],
+    #                      names=("wave_low_um", "wave_mid_um", "wave_upp_um", "RpRs2_ppm", "err_low_ppm", "err_upp_ppm"))
+    # else:
+        # t_spec = None
```
I would delete all these commented out chunks of code unless @kevin218 disagrees. Having large commented out blocks of code makes things hard to read through in the future. If ever we need the code, it is already saved in the Git history

cpiaulet replied:
I'll do it if Kevin thinks that's the best way to go, I left them there because I thought it was my only instruction :)

cpiaulet replied:
done

taylorbell57 commented on eureka/S5_lightcurve_fitting/fitters.py:
```diff
@@ -928,17 +1011,33 @@ def load_old_fitparams(meta, log, channel, freenames):
-        data = np.append(data, -lower_errs.reshape(1,-1), axis=0)
-        data = np.append(data, upper_errs.reshape(1,-1), axis=0)
-    np.savetxt(meta.outputdir+fname+'.csv', data, header=','.join(freenames), delimiter=',')
+    results_table.write(meta.outputdir+fname+'.csv', format='csv', overwrite=False)
```
Given that the file format for the S5 file has changed, you will also need to change the way in which that file is read in by S6

cpiaulet replied:
done now!

cpiaulet replied:
done now!

taylorbell57 commented:
Can you please go through your edits and make sure that they are all flake8 compliant? If you're using an editor like VS Code, you can make flake8 check your files continuously while you edit them. Otherwise, you can use flake8 in the command line after pip installing it. Sticking to a standard like flake8 will make sure that all of our code is nice and uniformly formatted. And making sure that all our PRs are all consistent with flake8 is important otherwise someone has to go through and edit every file at some point to make sure they're all compliant (I did this a week ago and it wasn't fun).

I ran the following command in the main "Eureka" folder and got many warnings/errors that are introduced with your edits:
`flake8 --exclude="./build/**/*,./docs/**/*,.vscode/*,**/site-packages/**/*,*.c,*.pyx,**/*niriss*.py" --per-file-ignores="**/__init__.py:F401,**/manageevent.py:F841,./demos/**/run_*.py:E402,**/test_*.py:E402,*.py:E722" .`

The following are the errors/warnings that were introduced with your edits:
```
./eureka/S6_planet_spectra/s6_spectra.py:352:80: E501 line too long (99 > 79 characters)
./eureka/S6_planet_spectra/s6_spectra.py:356:13: E303 too many blank lines (2)
./eureka/S6_planet_spectra/s6_spectra.py:381:63: W291 trailing whitespace
./eureka/S6_planet_spectra/s6_spectra.py:382:80: E501 line too long (99 > 79 characters)
./eureka/S6_planet_spectra/s6_spectra.py:384:23: E225 missing whitespace around operator
./eureka/S6_planet_spectra/s6_spectra.py:388:25: E225 missing whitespace around operator
./eureka/S6_planet_spectra/s6_spectra.py:389:80: E501 line too long (116 > 79 characters)
./eureka/S6_planet_spectra/s6_spectra.py:390:37: E128 continuation line under-indented for visual indent
./eureka/S6_planet_spectra/s6_spectra.py:390:38: E225 missing whitespace around operator
./eureka/S6_planet_spectra/s6_spectra.py:395:5: E303 too many blank lines (2)
./eureka/S5_lightcurve_fitting/fitters.py:115:1: W293 blank line contains whitespace
./eureka/S5_lightcurve_fitting/fitters.py:117:53: W291 trailing whitespace
./eureka/S5_lightcurve_fitting/fitters.py:118:57: E202 whitespace before ')'
./eureka/S5_lightcurve_fitting/fitters.py:118:59: W291 trailing whitespace
./eureka/S5_lightcurve_fitting/fitters.py:138:80: E501 line too long (90 > 79 characters)
./eureka/S5_lightcurve_fitting/fitters.py:278:80: E501 line too long (87 > 79 characters)
./eureka/S5_lightcurve_fitting/fitters.py:362:22: E261 at least two spaces before inline comment
./eureka/S5_lightcurve_fitting/fitters.py:366:63: E231 missing whitespace after ','
./eureka/S5_lightcurve_fitting/fitters.py:366:73: E231 missing whitespace after ','
./eureka/S5_lightcurve_fitting/fitters.py:366:78: E231 missing whitespace after ','
./eureka/S5_lightcurve_fitting/fitters.py:366:80: E501 line too long (96 > 79 characters)
./eureka/S5_lightcurve_fitting/fitters.py:366:97: W291 trailing whitespace
./eureka/S5_lightcurve_fitting/fitters.py:367:80: E501 line too long (102 > 79 characters)
./eureka/S5_lightcurve_fitting/fitters.py:371:1: W293 blank line contains whitespace
./eureka/S5_lightcurve_fitting/fitters.py:390:80: E501 line too long (90 > 79 characters)
./eureka/S5_lightcurve_fitting/fitters.py:395:5: E303 too many blank lines (3)
./eureka/S5_lightcurve_fitting/fitters.py:763:38: W291 trailing whitespace
./eureka/S5_lightcurve_fitting/fitters.py:765:80: E501 line too long (87 > 79 characters)
./eureka/S5_lightcurve_fitting/fitters.py:830:5: E303 too many blank lines (2)
./eureka/S5_lightcurve_fitting/fitters.py:832:22: E261 at least two spaces before inline comment
./eureka/S5_lightcurve_fitting/fitters.py:834:1: W293 blank line contains whitespace
./eureka/S5_lightcurve_fitting/fitters.py:836:63: E231 missing whitespace after ','
./eureka/S5_lightcurve_fitting/fitters.py:836:73: E231 missing whitespace after ','
./eureka/S5_lightcurve_fitting/fitters.py:836:80: E501 line too long (97 > 79 characters)
./eureka/S5_lightcurve_fitting/fitters.py:836:98: W291 trailing whitespace
./eureka/S5_lightcurve_fitting/fitters.py:837:80: E501 line too long (102 > 79 characters)
./eureka/S5_lightcurve_fitting/fitters.py:838:1: W293 blank line contains whitespace
./eureka/S5_lightcurve_fitting/fitters.py:858:1: W293 blank line contains whitespace
./eureka/S5_lightcurve_fitting/fitters.py:860:80: E501 line too long (94 > 79 characters)
./eureka/S5_lightcurve_fitting/fitters.py:864:5: E303 too many blank lines (2)
./eureka/S5_lightcurve_fitting/fitters.py:869:5: E303 too many blank lines (2)
./eureka/S5_lightcurve_fitting/fitters.py:954:38: W291 trailing whitespace
./eureka/S5_lightcurve_fitting/fitters.py:989:53: W291 trailing whitespace
./eureka/S5_lightcurve_fitting/fitters.py:990:57: E202 whitespace before ')'
./eureka/S5_lightcurve_fitting/fitters.py:990:59: W291 trailing whitespace
./eureka/S5_lightcurve_fitting/fitters.py:991:1: W293 blank line contains whitespace
./eureka/S5_lightcurve_fitting/fitters.py:993:5: E303 too many blank lines (2)
./eureka/S5_lightcurve_fitting/fitters.py:1011:80: E501 line too long (84 > 79 characters)
./eureka/S5_lightcurve_fitting/fitters.py:1015:5: E303 too many blank lines (2)
./eureka/S5_lightcurve_fitting/fitters.py:1226:80: E501 line too long (93 > 79 characters)
./eureka/S5_lightcurve_fitting/fitters.py:1247:1: W293 blank line contains whitespace
./eureka/S5_lightcurve_fitting/fitters.py:1252:80: E501 line too long (87 > 79 characters)
./eureka/S5_lightcurve_fitting/fitters.py:1258:80: E501 line too long (89 > 79 characters)
./eureka/S5_lightcurve_fitting/fitters.py:1259:80: E501 line too long (83 > 79 characters)
./eureka/S5_lightcurve_fitting/fitters.py:1262:5: E303 too many blank lines (2)
./eureka/S5_lightcurve_fitting/fitters.py:1267:46: F821 undefined name 'ch_number'
./eureka/S5_lightcurve_fitting/fitters.py:1276:29: F821 undefined name 'ch_number'
```

The following are errors that already existend and you don't need to worry about:
```
./eureka/S3_data_reduction/background.py:338:41: F821 undefined name 'mask2'
./tests/test_MIRI.py:12:5: F401 'eureka.S1_detector_processing.s1_process as s1' imported but unused
./demos/JWST/run_eureka.py:5:1: F401 'eureka.S1_detector_processing.s1_process as s1' imported but unused
```

It is also important that you check that your PRs pass all of the tests already written and that you write new tests if you've added a significantly new feature that isn't covered by previous tests (new tests aren't needed for this PR). You can check if your PR passes the current tests by running `pytest tests` in the main "Eureka" folder (you may need to pip install `pytest` if you haven't already installed it). Pytest also picks up on one of the same errors flake8 found which is that there is now an undefined variable `ch_number` in the function save_fit inside the fitters.py file.

Finally, also note that in the docstrings for functions, there needs to be a blank line after "History:" otherwise the docs won't be properly formatted (annoying, I know).

cpiaulet commented:
> Can you please go through your edits and make sure that they are all flake8 compliant? If you're using an editor like VS Code, you can make flake8 check your files continuously while you edit them. Otherwise, you can use flake8 in the command line after pip installing it. Sticking to a standard like flake8 will make sure that all of our code is nice and uniformly formatted. And making sure that all our PRs are all consistent with flake8 is important otherwise someone has to go through and edit every file at some point to make sure they're all compliant (I did this a week ago and it wasn't fun).
> 
> I ran the following command in the main "Eureka" folder and got many warnings/errors that are introduced with your edits: `flake8 --exclude="./build/**/*,./docs/**/*,.vscode/*,**/site-packages/**/*,*.c,*.pyx,**/*niriss*.py" --per-file-ignores="**/__init__.py:F401,**/manageevent.py:F841,./demos/**/run_*.py:E402,**/test_*.py:E402,*.py:E722" .`
> 
> The following are the errors/warnings that were introduced with your edits:
> 
> ```
> ./eureka/S6_planet_spectra/s6_spectra.py:352:80: E501 line too long (99 > 79 characters)
> ./eureka/S6_planet_spectra/s6_spectra.py:356:13: E303 too many blank lines (2)
> ./eureka/S6_planet_spectra/s6_spectra.py:381:63: W291 trailing whitespace
> ./eureka/S6_planet_spectra/s6_spectra.py:382:80: E501 line too long (99 > 79 characters)
> ./eureka/S6_planet_spectra/s6_spectra.py:384:23: E225 missing whitespace around operator
> ./eureka/S6_planet_spectra/s6_spectra.py:388:25: E225 missing whitespace around operator
> ./eureka/S6_planet_spectra/s6_spectra.py:389:80: E501 line too long (116 > 79 characters)
> ./eureka/S6_planet_spectra/s6_spectra.py:390:37: E128 continuation line under-indented for visual indent
> ./eureka/S6_planet_spectra/s6_spectra.py:390:38: E225 missing whitespace around operator
> ./eureka/S6_planet_spectra/s6_spectra.py:395:5: E303 too many blank lines (2)
> ./eureka/S5_lightcurve_fitting/fitters.py:115:1: W293 blank line contains whitespace
> ./eureka/S5_lightcurve_fitting/fitters.py:117:53: W291 trailing whitespace
> ./eureka/S5_lightcurve_fitting/fitters.py:118:57: E202 whitespace before ')'
> ./eureka/S5_lightcurve_fitting/fitters.py:118:59: W291 trailing whitespace
> ./eureka/S5_lightcurve_fitting/fitters.py:138:80: E501 line too long (90 > 79 characters)
> ./eureka/S5_lightcurve_fitting/fitters.py:278:80: E501 line too long (87 > 79 characters)
> ./eureka/S5_lightcurve_fitting/fitters.py:362:22: E261 at least two spaces before inline comment
> ./eureka/S5_lightcurve_fitting/fitters.py:366:63: E231 missing whitespace after ','
> ./eureka/S5_lightcurve_fitting/fitters.py:366:73: E231 missing whitespace after ','
> ./eureka/S5_lightcurve_fitting/fitters.py:366:78: E231 missing whitespace after ','
> ./eureka/S5_lightcurve_fitting/fitters.py:366:80: E501 line too long (96 > 79 characters)
> ./eureka/S5_lightcurve_fitting/fitters.py:366:97: W291 trailing whitespace
> ./eureka/S5_lightcurve_fitting/fitters.py:367:80: E501 line too long (102 > 79 characters)
> ./eureka/S5_lightcurve_fitting/fitters.py:371:1: W293 blank line contains whitespace
> ./eureka/S5_lightcurve_fitting/fitters.py:390:80: E501 line too long (90 > 79 characters)
> ./eureka/S5_lightcurve_fitting/fitters.py:395:5: E303 too many blank lines (3)
> ./eureka/S5_lightcurve_fitting/fitters.py:763:38: W291 trailing whitespace
> ./eureka/S5_lightcurve_fitting/fitters.py:765:80: E501 line too long (87 > 79 characters)
> ./eureka/S5_lightcurve_fitting/fitters.py:830:5: E303 too many blank lines (2)
> ./eureka/S5_lightcurve_fitting/fitters.py:832:22: E261 at least two spaces before inline comment
> ./eureka/S5_lightcurve_fitting/fitters.py:834:1: W293 blank line contains whitespace
> ./eureka/S5_lightcurve_fitting/fitters.py:836:63: E231 missing whitespace after ','
> ./eureka/S5_lightcurve_fitting/fitters.py:836:73: E231 missing whitespace after ','
> ./eureka/S5_lightcurve_fitting/fitters.py:836:80: E501 line too long (97 > 79 characters)
> ./eureka/S5_lightcurve_fitting/fitters.py:836:98: W291 trailing whitespace
> ./eureka/S5_lightcurve_fitting/fitters.py:837:80: E501 line too long (102 > 79 characters)
> ./eureka/S5_lightcurve_fitting/fitters.py:838:1: W293 blank line contains whitespace
> ./eureka/S5_lightcurve_fitting/fitters.py:858:1: W293 blank line contains whitespace
> ./eureka/S5_lightcurve_fitting/fitters.py:860:80: E501 line too long (94 > 79 characters)
> ./eureka/S5_lightcurve_fitting/fitters.py:864:5: E303 too many blank lines (2)
> ./eureka/S5_lightcurve_fitting/fitters.py:869:5: E303 too many blank lines (2)
> ./eureka/S5_lightcurve_fitting/fitters.py:954:38: W291 trailing whitespace
> ./eureka/S5_lightcurve_fitting/fitters.py:989:53: W291 trailing whitespace
> ./eureka/S5_lightcurve_fitting/fitters.py:990:57: E202 whitespace before ')'
> ./eureka/S5_lightcurve_fitting/fitters.py:990:59: W291 trailing whitespace
> ./eureka/S5_lightcurve_fitting/fitters.py:991:1: W293 blank line contains whitespace
> ./eureka/S5_lightcurve_fitting/fitters.py:993:5: E303 too many blank lines (2)
> ./eureka/S5_lightcurve_fitting/fitters.py:1011:80: E501 line too long (84 > 79 characters)
> ./eureka/S5_lightcurve_fitting/fitters.py:1015:5: E303 too many blank lines (2)
> ./eureka/S5_lightcurve_fitting/fitters.py:1226:80: E501 line too long (93 > 79 characters)
> ./eureka/S5_lightcurve_fitting/fitters.py:1247:1: W293 blank line contains whitespace
> ./eureka/S5_lightcurve_fitting/fitters.py:1252:80: E501 line too long (87 > 79 characters)
> ./eureka/S5_lightcurve_fitting/fitters.py:1258:80: E501 line too long (89 > 79 characters)
> ./eureka/S5_lightcurve_fitting/fitters.py:1259:80: E501 line too long (83 > 79 characters)
> ./eureka/S5_lightcurve_fitting/fitters.py:1262:5: E303 too many blank lines (2)
> ./eureka/S5_lightcurve_fitting/fitters.py:1267:46: F821 undefined name 'ch_number'
> ./eureka/S5_lightcurve_fitting/fitters.py:1276:29: F821 undefined name 'ch_number'
> ```
> 
> The following are errors that already existed and you don't need to worry about:
> 
> ```
> ./eureka/S3_data_reduction/background.py:338:41: F821 undefined name 'mask2'
> ./tests/test_MIRI.py:12:5: F401 'eureka.S1_detector_processing.s1_process as s1' imported but unused
> ./demos/JWST/run_eureka.py:5:1: F401 'eureka.S1_detector_processing.s1_process as s1' imported but unused
> ```
> 
> It is also important that you check that your PRs pass all of the tests already written and that you write new tests if you've added a significantly new feature that isn't covered by previous tests (new tests aren't needed for this PR). You can check if your PR passes the current tests by running `pytest tests` in the main "Eureka" folder (you may need to pip install `pytest` if you haven't already installed it). Pytest also picks up on one of the same errors flake8 found which is that there is now an undefined variable `ch_number` in the function save_fit inside the fitters.py file.
> 
> Finally, also note that in the docstrings for functions, there needs to be a blank line after "History:" otherwise the docs won't be properly formatted (annoying, I know).

Alright, my latest commit should have fixed all of these. Let me know if anything else is needed!

taylorbell57 commented:
There are still a few warnings/errors with flake8 (use the command I gave above to see them).

There are also now two new errors related to stage 6 with the pytest command. For NIRCam, the following line in the parse_s5_saves function throws an error whose message suggests that fitted_values has no such element "50th":
```
medians = np.array(fitted_values["50th"])
```

For the MIRI test, the np.percentile call on line 375 of s6_spectra.py (also in the same parse_s5_saves function) gives an index error that I'm not quite sure the cause of. Since these percentiles are already saved in your csv file though, why not just use those rather than recomputing them?

cpiaulet commented:
> There are still a few warnings/errors with flake8 (use the command I gave above to see them).
> 
> There are also now two new errors related to stage 6 with the pytest command. For NIRCam, the following line in the parse_s5_saves function throws an error whose message suggests that fitted_values has no such element "50th":
> 
> ```
> medians = np.array(fitted_values["50th"])
> ```
> 
> For the MIRI test, the np.percentile call on line 375 of s6_spectra.py (also in the same parse_s5_saves function) gives an index error that I'm not quite sure the cause of. Since these percentiles are already saved in your csv file though, why not just use those rather than recomputing them?

For your first point, I fixed all the remaining flake8 issues except the following: 
```
./eureka/S3_data_reduction/background.py:338:41: F821 undefined name 'mask2'
./tests/test_MIRI.py:12:5: F401 'eureka.S1_detector_processing.s1_process as s1' imported but unused
```
which are not in the files I touched for this PR - but that I wanted to report. 

For the NIRCam test, the issue was happening because in the case of a lsq fit, there are no percentiles saved (because there are no samples to draw percentiles from). I added a statement so that it uses the lsq fit result in that case.

For the MIRI test, I implemented the use of the percentiles recorded in the csv file.

I have run the `flake8` check and `pytest` commands again with no issues, but please let me know if there is anything else I can do to make this merge easier :)  + have a good weekend!

taylorbell57 commented on eureka/S5_lightcurve_fitting/fitters.py:
```diff
@@ -1213,23 +1238,25 @@ def save_fit(meta, lc, model, fitter, fit_params, freenames, samples=[],
-        The one sigma upper limits from a sampling method, by default [].
-    lower_errs : ndarray; optional
-        The one sigma lower limits from a sampling method, by default [].
+    spec_table: astropy table; optional
+        Transmission spectrum (now only produced in S6)
```
Since spec_table is never used, let's remove it from the docstring

taylorbell57 commented on eureka/S5_lightcurve_fitting/fitters.py:
```diff
@@ -1213,23 +1238,25 @@ def save_fit(meta, lc, model, fitter, fit_params, freenames, samples=[],
+    History:
+
+    - Mar 13-Apr 18, 2022 Caroline Piaulet
+         Record an astropy table for mean, median, percentiles,
+             +/- 1 sigma, all params
+         Save transmission spectrum (commented out)
```
This history line (and the similar one for the emceefitter function) are indented by one too many spaces (should only be 4 spaces ahead of the start of the dash), and you can't have the "+/- ..." line be extra indented; both of these cause warnings while building the docs. "Save transmission spectrum (commented out)" can also be removed from both history lines.

taylorbell57 reviewed (changes requested):
Alright, just a few tiny tweaks related to the docstrings and then this PR is good to go!

FYI, I'm hoping to make built-in tests on GitHub for flake8, pytest, and ideally building the docs so that we can all see the results of the tests right on the PR page, but for now everything needs to be checked manually

taylorbell57 commented:
I just realized I'm able to push changes to your PR (I couldn't before, but I was using a different method which didn't work since I don't have write privs for your fork), so I made the tiny tweaks that I'd requested. As long as you're happy with the edits I made, you can now click merge below
